FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -157,28 +159,33 @@ - (void)signInTesterWithCompletion:(void (^)(NSError *_Nullable error))completio
 
     FIRFADDebugLog(@"Registration URL: %@", requestURL);
 
-    [self.appDelegateInterceptor
+    [self.UIService
         appDistributionRegistrationFlow:[[NSURL alloc] initWithString:requestURL]
                          withCompletion:^(NSError *_Nullable error) {
                            FIRFADInfoLog(@"Tester sign in complete.");
-                           if (!error) {
-                             [self persistTesterSignInState];
+                           if (error) {
+                             completion(error);
+                             return;
                            }
-                           completion(error);
+                           [self persistTesterSignInStateAndHandleCompletion:completion];
                          }];
   }];
 }
 
-- (void)persistTesterSignInState {
-  [FIRFADApiService
-      fetchReleasesWithCompletion:^(NSArray *_Nullable releases, NSError *_Nullable error) {
-        if (error) {
-          FIRFADErrorLog(@"Could not fetch releases with code %ld - %@", [error code],
-                         [error localizedDescription]);
-          return;
-        }
-        [[GULUserDefaults standardUserDefaults] setBool:YES forKey:kFIRFADSignInStateKey];
-      }];
+- (void)persistTesterSignInStateAndHandleCompletion:(void (^)(NSError *_Nullable error))completion {
+  [FIRFADApiService fetchReleasesWithCompletion:^(NSArray *_Nullable releases,
+                                                  NSError *_Nullable error) {
+    if (error) {
+      FIRFADErrorLog(@"Could not fetch releases with code %ld - %@", [error code],
+                     [error localizedDescription]);
+      completion([self NSErrorForErrorCodeAndMessage:FIRAppDistributionErrorAuthenticationFailure
```

Comment:
hmmm... Do we want this to be an authentication error? I think you want to call the error mapper -https://github.com/firebase/firebase-ios-sdk/blob/app-distribution-eap-reloaded/FirebaseAppDistribution/Sources/FIRAppDistribution.m#L211 - to map the error codes returned from the fetch call appropriately.

Reply:
I think this is an authentication error and not a fetch releases error. We call fetch releases because it's the only way to validate if the registration is successful. But the end result for the user is that registration failed and the error returned back should reflect that.

Reply:
The error here is really that we failed to persist the sign in state because we failed to fetch releases, but we could have failed to fetch releases because of a network timeout, and that is meaningfully different than a failure to authenticate. In the case where we have successfully registered but failed to persist that state, a call to fetch may subsequently succeed, we just won't know about it session over session. Imo the more specific we can be with the error, the better.